### PR TITLE
Cashflow.js -> Cashflow.tsx

### PR DESCRIPTION
**Pull request by lucasboebel**
first pull request, feedback appreciated :)

### CODE REVIEW

**Comment by MatissJanis** [comment]
👋 Just checking in: would you like this PR to be reviewed? It's currently in draft, so double checking.

File: packages/desktop-client/src/components/reports/graphs/CashFlowGraph.tsx [hunk 1]
```diff
@@ -16,10 +16,17 @@ import Container from '../Container';
...
 type CashFlowGraphProps = {
+  start: string;
+  end: string;
   graphData: { expenses; income; balances };
   isConcise: boolean;
 };
-function CashFlowGraph({ graphData, isConcise }: CashFlowGraphProps) {
+function CashFlowGraph({
+  start,
+  end,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/CashFlowGraph.tsx:26]: **💬 suggestion:** ‏the component is not using `start`/`end` props. As such: we should not add them.

File: packages/desktop-client/src/components/reports/graphs/CashFlowGraph.tsx [hunk 2]
```diff
@@ -16,10 +16,17 @@ import Container from '../Container';
 import Tooltip from '../Tooltip';
 
 type CashFlowGraphProps = {
+  start: string;
+  end: string;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/CashFlowGraph.tsx:20]: Same thing here

File: packages/desktop-client/src/components/reports/reports/CashFlow.tsx [hunk 3]
```diff
@@ -52,7 +52,8 @@ function CashFlow() {
 
   useEffect(() => {
     async function run() {
-      const trans = await send('get-earliest-transaction');
+      let trans;
+      trans = await send('get-earliest-transaction');
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/reports/CashFlow.tsx:56]: **🔨 warning:** ‏previous implementation here was much better. What's the problem you're experiencing? There must be a different solution to it.

**Review by MatissJanis — CHANGES_REQUESTED** [review]

File: packages/desktop-client/src/components/reports/reports/CashFlow.tsx (hunk 3 above)
Comment (lucasboebel) [reply packages/desktop-client/src/components/reports/reports/CashFlow.tsx:56]: I'm seeing "Property 'date' does not exist on type 'unknown'." on line 58 with 'const'. 'let' removed the error but not sure if there is a better way to do it

**Comment by youngcw** [comment]
Can you look into why the tests are failing?

File: packages/desktop-client/src/components/reports/reports/CashFlow.tsx (hunk 3 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/reports/reports/CashFlow.tsx:56]: You'll want to update [this line](https://github.com/actualbudget/actual/blob/b3bcff094db3d55445325357d943d1377181c808/packages/loot-core/src/types/server-handlers.d.ts#L57) to solve that problem. From `unknown` to a full-er datatype.

**Pull request merged**